### PR TITLE
kb-handling: Refactor kb files management

### PR DESCRIPTION
**Pull request by zzacharo**
* Unifies kb file handling for all kb builders

Signed-off-by: Zacharias Zacharodimos <zacharias.zacharodimos@cern.ch>

### CODE REVIEW

**Comment by zzacharo** [comment]
@kaplun @jacquerie I did some refactoring to kbs file handling! Check it please.

File: tests/test_api.py [hunk 1]
```diff
@@ -121,41 +121,42 @@ def test_journal_extract():
...
-        url = "http://www.example.com"
-        responses.add(
-            responses.GET,
-            url,
-            body="File not found!",
-            status=404,
-            content_type='text/plain',
-        )
-        extract_references_from_url(url)
+# def test_extract_references_from_string(kbs_override):
```
Comment (kaplun) [inline tests/test_api.py]: Why commenting out the tests?

Comment (zzacharo) [inline tests/test_api.py]: oups leftover!

Comment (zzacharo) [inline tests/test_api.py]: I am gonna push  again!

**Comment by zzacharo** [comment]
@kaplun done!

**Comment by kaplun** [comment]
Looks like Travis is bad on Python 2.6, but since we don't intend to support Python 2.6 anymore (only legacy is on 2.6), can you add a commit where you drop Python 2.6 from the Travis build matrix and setup.py (if it's there)?

**Comment by zzacharo** [comment]
@kaplun done!

**Comment by kaplun** [comment]
Yep, can you also amend indeed: https://github.com/inspirehep/refextract/blob/master/setup.py#L121 (just leave the 2.7 line, because since we are not currently testing on 3.x we can't guarantee it's actually supported).

File: refextract/references/kbs.py [hunk 2]
```diff
@@ -42,6 +43,23 @@
 from ..documents.text import re_group_captured_multiple_space
 
 
+@contextlib.contextmanager
+def file_resolving(fpath, reader=None, **kwargs):
+    if isinstance(fpath, six.string_types):
+        print('Load kb path %s' % fpath)
```
Comment (jacquerie) [inline refextract/references/kbs.py]: IMO we should think about removing all this printing from `refextract`. The argument is basically the "Rule of Silence" by Eric S. Raymond (https://en.wikipedia.org/wiki/Unix_philosophy#Eric_Raymond.E2.80.99s_17_Unix_Rules)
> Developers should design programs so that they do not print unnecessary output. This rule aims to allow other programs and developers to pick out the information they need from a program's output without having to parse verbosity.

File: refextract/references/kbs.py [hunk 3]
```diff
@@ -42,6 +43,23 @@
...
+        fh = codecs.open(fpath, encoding='utf-8')
+        if reader:
+            yield reader(fh, delimiter=kwargs.pop('delimiter', b'|'), lineterminator=kwargs.pop('lineterminator', ';'))
+        else:
+            yield fh
+
+        print('Close file with kb path %s' % fpath)
```
Comment (jacquerie) [inline refextract/references/kbs.py]: Same as above, and for the rest of the PR.

File: refextract/references/kbs.py (hunk 2 above)
Comment (kaplun) [inline refextract/references/kbs.py]: I agree, although, in this PR @zzacharo was just refactoring. So moving to use the logging library would be the right thing to do, (either in a separate PR or at least a separate commit).

Note however that print is many more files...

Comment (zzacharo) [inline refextract/references/kbs.py]: Yep I left them because I didn't know if they are needed! I think IOError message is quite descriptive when i/o exception is occured! So I will clean them ok?

File: .gitignore [hunk 4]
```diff
@@ -1,3 +1,4 @@
+.idea/
```
Comment (jacquerie) [inline .gitignore]: Ok, but unrelated: should live in its own commit.

Comment (zzacharo) [inline .gitignore]: I am gonna split it in another commit.

File: refextract/references/kbs.py (hunk 2 above)
Comment (kaplun) [inline refextract/references/kbs.py]: They are actually not expeceted by any code. They are only useful when using refextract by CLI, for debugging purposes. But then indeed, would be great to look through all refextract and change all the prints for the proper logging.debug/info/warning... commands.

Comment (zzacharo) [inline refextract/references/kbs.py]: So, you mean introducing a proper logger for the needed messages?

Comment (jacquerie) [inline refextract/references/kbs.py]: > I agree, although, in this PR @zzacharo was just refactoring.

True for this particular `print`, not true for all the other `print`s that he's adding below in the PR... I'd delete those. And this, honestly...

Comment (jacquerie) [inline refextract/references/kbs.py]: BTW it's ok when refactoring to remove non functional (as in: not part of the API or output that other collaborating programs expect) and debugging code.

Comment (zzacharo) [inline refextract/references/kbs.py]: @jacquerie I agree and there cases here that can be removed. So, I will leave only a logger message inside the contextmanager and I will delete all the other messages. And if more log messages are needed, I can create another PR to introduce them.

Comment (zzacharo) [inline refextract/references/kbs.py]: @jacquerie https://github.com/zzacharo/refextract/blob/e1dc83392e37698f206d27144eb57db9a04dd65c/refextract/references/kbs.py#L487 this print how should be handled?

**Comment by zzacharo** [comment]
@jacquerie @kaplun have a look again. I deleted all the print functions for now as they weren't make sense because exceptions are implying that our code is wrong so we must get the exception and fix the root problem. If a logger is needed in the future we can add it in another PR.

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+<component name="InspectionProjectProfileManager">
+  <settings>
+    <option name="useProjectProfile" value="false" />
+    <option name="USE_PROJECT_PROFILE" value="false" />
+    <version value="1.0" />
+  </settings>
+</component>
```
Comment (kaplun) [inline .idea/inspectionProfiles/profiles_settings.xml]: Leftover :)

**Comment by zzacharo** [comment]
@kaplun done!
